### PR TITLE
NH-102467 Python: Add deducing of export mode from SW_APM_COLLECTOR

### PR DESCRIPTION
**Pull request by jerrytfleung**
Added deducing mode from SW_APM_COLLECTOR

https://swicloud.atlassian.net/browse/NH-102467

### CODE REVIEW

File: tests/unit/test_distro.py [hunk 1]
```diff
@@ -346,6 +346,78 @@ def test_configure_env_endpoint(self, mocker):
...
+        )
+        distro.SolarWindsDistro()._configure()
+        assert os.environ[OTEL_EXPORTER_OTLP_ENDPOINT] == "https://otel.collector.na-02.st-ssp.solarwinds.com:443"
+
+    def test_configure_env_service_key_and_collector_only_invalid(self, mocker):
+        mocker.patch.dict(
+            os.environ,
+            {
+                "SW_APM_SERVICE_KEY": "foo-token:bar",
+                "SW_APM_COLLECTOR": "www.google.com",
```
Comment (tammy-baylis-swi) [inline tests/unit/test_distro.py:401]: This one I am not sure about, and apologies because it's not in ticket OP either.

@cheempz , does this make sense for this iteration?: If `SW_APM_COLLECTOR` is anything but `apm.collector`, then APM uses the `otel.collector.na-01` default. OR, APM uses what's configured and lets settings collection and export fail for visible error.

File: tests/unit/test_distro.py [hunk 2]
```diff
@@ -346,6 +346,78 @@ def test_configure_env_endpoint(self, mocker):
...
+            },
+            clear=True
+        )
+        distro.SolarWindsDistro()._configure()
+        assert os.environ[OTEL_EXPORTER_OTLP_ENDPOINT] == "https://otel.collector.na-01.cloud.solarwinds.com:443"
+
```
Comment (tammy-baylis-swi) [inline tests/unit/test_distro.py:407]: Weird ask because it's repetitive but it'll help with spec clarity:

Could we have an additional unit test where `"SW_APM_COLLECTOR": "otel.collector.na-02.st-ssp.solarwinds.com"` then the assert is the endpoint is the default `"https://otel.collector.na-01.cloud.solarwinds.com:443"`. This is to make clear that `COLLECTOR` doesn't get resolved if `otel.collector` which I _think_ is how we address that edge case.

**Comment by tammy-baylis-swi** [comment]
Thank you! Overall looks good but I'd like a 2nd set of eyes on this before merging since it relates to this week's conversations 🙂

File: tests/unit/test_distro.py (hunk 2 above)
Comment (jerrytfleung) [reply tests/unit/test_distro.py:419]: Sure. Done

File: tests/unit/test_distro.py (hunk 1 above)
Comment (jerrytfleung) [reply tests/unit/test_distro.py:401]: My thinking was to greedily extract `xx-yy.<cloud/st-ssp/dev-ssp>` from the environment variable and substitute to `https://otel.collector.<substitute>.solarwinds.com:443`

Let me know if it doesn't making sense. Thanks!

**Review by tammy-baylis-swi — APPROVED** [review]

File: tests/unit/test_distro.py (hunk 1 above)
Comment (cheempz) [reply tests/unit/test_distro.py:401]: yes it makes sense to use the fallback value if SW APM config for collector is anything other than `apm.collector.*`

File: tests/unit/test_distro.py [hunk 3]
```diff
@@ -346,6 +346,90 @@ def test_configure_env_endpoint(self, mocker):
...
+    def test_configure_env_service_key_and_collector_and_otel_exporter(self, mocker):
+        mocker.patch.dict(
+            os.environ,
+            {
+                "SW_APM_SERVICE_KEY": "foo-token:bar",
+                "SW_APM_COLLECTOR": "apm.collector.na-02.cloud.solarwinds.com",
+                "OTEL_EXPORTER_OTLP_ENDPOINT": "https://apm.collector.na-03.cloud.solarwinds.com:443",
```
Comment (cheempz) [inline tests/unit/test_distro.py:426]: this is a good test that we respect explict OTEL exporter config as highest precedence. my nit is the value could be something that does not imply the apm.collector actually supports OTLP.

Comment (jerrytfleung) [reply tests/unit/test_distro.py]: Sure. Done
